crates/router/src/routes/admin.rs

```diff
@@ -383,6 +391,74 @@ pub async fn payment_connector_retrieve(
     )
     .await
 }
+/// Merchant Connector - Retrieve
+///
+/// Retrieve Merchant Connector Details
+#[cfg(all(feature = "v2", feature = "merchant_connector_account_v2"))]
+#[utoipa::path(
+    get,
+    path = "/connector_accounts/{id}",
+    params(
+        ("id" = i32, Path, description = "The unique identifier for the Merchant Connector")
+    ),
+    responses(
+        (status = 200, description = "Merchant Connector retrieved successfully", body = MerchantConnectorResponse),
+        (status = 404, description = "Merchant Connector does not exist in records"),
+        (status = 401, description = "Unauthorized request")
+    ),
+    tag = "Merchant Connector Account",
+    operation_id = "Retrieve a Merchant Connector",
+    security(("admin_api_key" = []))
+)]
+#[instrument(skip_all, fields(flow = ?Flow::MerchantConnectorsRetrieve))]
+pub async fn payment_connector_retrieve(
+    state: web::Data<AppState>,
+    req: HttpRequest,
+    path: web::Path<String>,
+) -> HttpResponse {
+    let flow = Flow::MerchantConnectorsRetrieve;
+    let id = path.into_inner();
+    let payload = web::Json(admin::MerchantConnectorId { id: id.clone() }).into_inner();
+
+    let x_merchant_id =
+        match auth::get_mandatory_header_value_by_key(headers::X_MERCHANT_ID.into(), req.headers())
+        {
+            Ok(val) => val.to_owned(),
+            Err(err) => {
+                return api::log_and_return_error_response(err);
+            }
+        };
+
+    let merchant_id =
+        match common_utils::id_type::MerchantId::try_from(std::borrow::Cow::from(x_merchant_id)) {
```

Comment:
this will the wrap function for MerchantId.

```
impl MerchantId {
    ...

    fn wrap(merchant_id: &str) -> Result<Cow<'_, Self>, _>

    ...
}

```